daemon/health.go

```diff
@@ -253,8 +253,14 @@ func (d *Daemon) initHealthMonitor(c *container.Container) {
 
 	if c.State.Health == nil {
 		h := &container.Health{}
-		h.Status = types.Starting
-		h.FailingStreak = 0
+
+		// Happens when the user specified NONE as the healthcheck.
+		if getProbe(c) == nil {
```

Comment:
I think we should move this check to the start of the function (replacing `if c.Config.Healthcheck == nil`). Containers that don't specify a healthcheck should behave the same as containers that specify no healthcheck. Then the `NoHealthCheck` state can be removed too.